src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.LocalFunctions.cs

```diff
@@ -163,9 +165,15 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
             {
                 this.State = pending.State;
                 BoundNode branch = pending.Branch;
-                LeaveParameters(localFuncSymbol.Parameters, branch?.Syntax,
-                                branch?.WasCompilerGenerated == true
-                                    ? location : null);
+
+                // No need to check the parameters if we dont know the branch
+                if (branch != null)
```

Comment:
It looks like the root cause isn't that the branch is null, it's that `LeaveParameters` doesn't expect the syntax to be null. I see two options:

1. We could harden LeaveParameters against null syntax.
2. We could find some syntax to pass to LeaveParameters in the case that the branch is null.

I think (2) may be our best option here. It seems like the syntax is just a place to report the diagnostic. Since this is a generated branch which implicitly occurs before the first statement in an iterator, it seems like the local function itself is as good as any to put the diagnostic (since there is no line of source code that actually corresponds to the implicit branch).

I would instead remove the initial null check and change `branch?.Syntax` below to `branch?.Syntax ?? localFunc.Syntax` and add a comment about why we're choosing the local function syntax in the case of null.